src/Surfnet/StepupMiddleware/CommandHandlingBundle/Identity/Service/RegistrationMailService.php

```diff
@@ -79,12 +122,75 @@ public function __construct(
         $this->emailTemplateService = $emailTemplateService;
         $this->fallbackLocale = $fallbackLocale;
         $this->selfServiceUrl = $selfServiceUrl;
+        $this->institutionConfigurationOptionsService = $institutionConfigurationOptionsService;
+        $this->identityService = $identityService;
+        $this->secondFactorService = $secondFactorService;
+        $this->raLocationsService = $raLocationService;
+        $this->raListingService = $raListingService;
+        $this->logger = $logger;
     }
 
-    /**
-     * @throws TransportExceptionInterface
-     */
-    public function sendRegistrationEmailWithRas(
+    public function send(string $identityId, string $secondFactorId)
+    {
+        $this->logger->notice(sprintf('Start processing of a registration email for %s', $identityId));
+        $identity = $this->identityService->find($identityId);
+        $institution = new Institution($identity->institution->getInstitution());
+        $institutionConfigurationOptions = $this->institutionConfigurationOptionsService
+            ->findInstitutionConfigurationOptionsFor($institution);
+        $verifiedSecondFactor = $this->secondFactorService->findVerified(new SecondFactorId($secondFactorId));
+
+        if ($institutionConfigurationOptions->useRaLocationsOption->isEnabled()) {
+            $this->logger->notice('Sending a registration mail with ra locations');
+
+            $this->sendRegistrationEmailWithRaLocations(
```

Comment:
Perhaps it was a conscious choice but methods sendRegistrationEmailWithRas and sendRegistrationEmailWithRaLocations contain a lot of duplicate code and actually differ in 1 parameter. Perhaps the code could be made simpler and more readable.